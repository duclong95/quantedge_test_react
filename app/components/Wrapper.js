import React, { Component } from 'react';
import TblHeader from './TblHeader';
import TblData from './TblData';
import AppBar from 'material-ui/AppBar';
import { Tabs, Tab } from 'material-ui/Tabs';

const styles = {
    tab: {
        fontSize: 18,
        padding: 8,
        marginLeft: 10,
        marginRight: 10,
        fontWeight: 400,
    },
};
class Wrapper extends Component {

    constructor(props){
        super(props);
        this.state = {
            sortType: 'gain'
        }
    }

    sort(type) {
        this.setState({
            sortType: type
        })
    }

    render() {
        return (
            <div>
                <AppBar
                    title="A&P/ASX"
                    showMenuIconButton={false}>
                    <Tabs>
                        <Tab className="gain" style={styles.tab} onActive={this.sort.bind(this, 'gain')} label="TOP GAINERS" >
                        </Tab>
                        <Tab className="lose" style={styles.tab} onActive={this.sort.bind(this, 'lose')} label="TOP LOSERS" >
                        </Tab>
                    </Tabs>
                </AppBar>
                <TblHeader />
                <TblData className="data" sortType={this.state.sortType} />
            </div>
        )
    }
}
module.exports = Wrapper;