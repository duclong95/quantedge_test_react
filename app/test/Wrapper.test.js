import React from 'react'
import { shallow, mount } from 'enzyme'
import toJson from 'enzyme-to-json'
import MuiThemeProvider from 'material-ui/styles/MuiThemeProvider'
import Wrapper from '../components/Wrapper'

const wrapper = shallow(
    <Wrapper />
);
test('Render Wrapper', ()=>{
    expect(toJson(wrapper)).toMatchSnapshot();
})
test('Top losers tab', () => {
    const tabLose = wrapper.find('.lose');
    tabLose.simulate('active');
    expect(wrapper.state().sortType).toEqual('lose')
});  
test('Top gainner tab', () => {
    const tabLose = wrapper.find('.gain');
    tabLose.simulate('active');
    expect(wrapper.state().sortType).toEqual('gain')
});  