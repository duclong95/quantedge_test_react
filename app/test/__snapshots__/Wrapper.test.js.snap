// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Wrapper 1`] = `
<div>
  <AppBar
    showMenuIconButton={false}
    title="A&P/ASX"
    zDepth={1}
  >
    <Tabs
      initialSelectedIndex={0}
      onChange={[Function]}
    >
      <Tab
        className="gain"
        label="TOP GAINERS"
        onActive={[Function]}
        style={
          Object {
            "fontSize": 18,
            "fontWeight": 400,
            "marginLeft": 10,
            "marginRight": 10,
            "padding": 8,
          }
        }
      />
      <Tab
        className="lose"
        label="TOP LOSERS"
        onActive={[Function]}
        style={
          Object {
            "fontSize": 18,
            "fontWeight": 400,
            "marginLeft": 10,
            "marginRight": 10,
            "padding": 8,
          }
        }
      />
    </Tabs>
  </AppBar>
  <TblHeader />
  <TblData
    className="data"
    sortType="gain"
  />
</div>
`;
